.github/workflows/app.yml

```
@@ -0,0 +1,36 @@
+name: Test and build Jellyfin Android
+on:
+  push:
+  release:
+    types: [ published ]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Cache Gradle wrapper and dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/wrapper/dists
+            ~/.gradle/caches
+          key: gradle-cache
+      - name: Set environment variable with version
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: |
+          echo 'JELLYFIN_VERSION<<EOF' >> $GITHUB_ENV
+          echo $GITHUB_REF | sed 's|refs/tags/||' >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
```

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable

```suggestion
        run:  echo "JELLYFIN_VERSION=${$GITHUB_REF | sed 's|refs/tags/||'}" >> $GITHUB_ENV
```
_not tested_

---

Alternatively you could set a output for this step, but it does not realy matter if you use step outputs or envs IMO

---

We need `JELLYFIN_VERSION` as environment variable in the Gradle build scripts so outputs would only make it more complicated because you'd basically do:
```
export JELLYFIN_VERSION = "${{ steps.variables.outputs.SomeVarName }}"
```
before running `gradlew`.

---

:man_facepalming: I forgot about that requirement, sorry about that

---

We do have some slightly unusual requirements in the build so I won't blame you for forgetting that haha

---

May I ask if it where possible to pass the `JELLYFIN_VERSION` as a gradle build propertie, just so its use is more transperant in the CI?
e.g. `./gradlew test assemble versionTxt -Pjellyfin.version=...`

---

I think that might help understanding people how the CI works so I'd say let's try it

---

I thought about doing this, but the problem is that you'd always set jellyfin.version then and would explicitly need to filter out e.g. empty strings. It makes the code more complicated as you need to specifically handle what the `if` in the env task did before, so I'd prefer to keep it as is.